File: an2_sem4/MPP/Homework/sapt10/cs/FestivalSellpoint/FestivalSellpoint.Service/IAppService.cs
using FestivalSellpoint.Domain;
using System.Collections.Generic;
using System;
using FestivalSellpoint.Service.Observer;

namespace FestivalSellpoint.Service
{
    public interface IAppService
    {
        Angajat LoginAngajat(string username, string password, IObserver client = null);
        void RegisterAngajat(string username, string password, string email);

        IEnumerable<Spectacol> GetAllSpectacole();

        IEnumerable<Spectacol> FilterSpectacole(DateTime startDate, DateTime endDate);

        IEnumerable<Spectacol> FilterSpectacole(DateTime day);

        void ReserveBilet(Spectacol spectacol, string cumparatorName, int seats);
    }
}

File: an2_sem4/MPP/Homework/sapt56/java/service/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/service/IAppService.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.service;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Angajat;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Spectacol;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.EntityRepoException;

import java.time.LocalDateTime;

public interface IAppService {
    Angajat loginAngajat(String username, String password) throws EntityRepoException;
    void registerAngajat(String username, String password, String email) throws EntityRepoException;

    Iterable<Spectacol> getAllSpectacole() throws EntityRepoException;

    Iterable<Spectacol> filterSpectacole(LocalDateTime startDate, LocalDateTime endDate) throws EntityRepoException;

    Iterable<Spectacol> filterSpectacole(LocalDateTime day) throws EntityRepoException;

    void reserveBilet(Spectacol spectacol, String cumparatorName, int seats) throws BiletReservationException, EntityRepoException;
}
